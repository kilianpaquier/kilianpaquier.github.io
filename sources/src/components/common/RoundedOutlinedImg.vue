<template>
  <div>
    <v-img
      :src="`/img/interests/${object.img}`"
      class="rounded-circle mx-auto mb-3 interests-border interests-img"
      v-animate-css.hover="'rubberBand'"
      eager
    />
    <span class="body-2" v-text="$vuetify.lang.t(`$vuetify.${object.title}`)" />
  </div>
</template>

<script lang="ts">
import Vue from "vue";
import Component from "vue-class-component";
import { Prop } from "vue-property-decorator";

@Component
export default class RoundedOutlinedImg extends Vue {
  @Prop({ required: true })
  object!: { img: string; title: string };
}
</script>

<style lang="sass" scoped>
.interests-border
    box-shadow: 0 0 0 7px #CFD8DC
    -moz-box-shadow: 0 0 0 7px #CFD8DC
    -o-box-shadow: 0 0 0 7px #CFD8DC
    -webkit-box-shadow: 0 0 0 7px #CFD8DC

.interests-img
    min-height: 90px
    min-width: 90px
    max-width: 90px
    max-height: 90px

</style>
