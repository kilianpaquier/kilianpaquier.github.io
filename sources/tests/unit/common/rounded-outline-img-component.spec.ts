import { createLocalVue, mount, Wrapper } from "@vue/test-utils";
import RoundedOutlinedImg from "@/components/common/RoundedOutlinedImg.vue";
import Vuetify from "vuetify";
import Vue from "vue";

Vue.use(Vuetify);
Vue.config.silent = true;
Vuetify.config.silent = true;

describe("RoundedOutlinedImg.vue", () => {
    const localVue = createLocalVue();

    let vuetify: Vuetify;

    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    let wrapper: Wrapper<any>;

    beforeEach(() => {
        vuetify = new Vuetify();

        wrapper = mount(RoundedOutlinedImg, {
            localVue,
            vuetify,
            propsData: {
                object: {
                    title: "title",
                    img: "movies.png",
                },
            },
        });
    });

    it("should create RoundedOutlinedImg", () => {
        expect(wrapper.html()).toMatchSnapshot();
    });
});
