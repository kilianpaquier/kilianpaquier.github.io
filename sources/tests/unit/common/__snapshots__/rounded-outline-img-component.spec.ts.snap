// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoundedOutlinedImg.vue should create RoundedOutlinedImg 1`] = `
<div>
  <div class="v-image v-responsive rounded-circle mx-auto mb-3 interests-border interests-img theme--light">
    <transition-stub name="fade-transition" mode="in-out">
      <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
    </transition-stub>
    <div class="v-responsive__content"></div>
  </div> <span class="body-2">$vuetify.title</span>
</div>
`;
